src/dlc-handlers/dfns-dlc-handler.ts

```diff
@@ -0,0 +1,369 @@
+  createTaprootMultisigPayment,
+  deriveUnhardenedPublicKey,
+  ecdsaPublicKeyToSchnorr,
+  getBalance,
+  getFeeRate,
+  getInputIndicesByScript,
+  getUnspendableKeyCommittedToUUID,
+} from '../functions/bitcoin/bitcoin-functions.js';
+import {
+  createDepositTransaction,
+  createFundingTransaction,
+  createWithdrawTransaction,
+} from '../functions/bitcoin/psbt-functions.js';
+import { PaymentInformation } from '../models/bitcoin-models.js';
+import { RawVault } from '../models/ethereum-models.js';
+
+export class DFNSDLCHandler {
+  private readonly dfnsDelegatedAPIClient: DfnsDelegatedApiClient;
+  private readonly bitcoinNetwork: Network;
+  private readonly bitcoinBlockchainAPI: string;
+  private readonly bitcoinBlockchainFeeRecommendationAPI: string;
+  private taprootDerivedPublicKey?: string;
+  private dfnsWalletID?: string;
+  public payment?: PaymentInformation;
+
+  constructor(
+    dfnsAppID: string,
+    appBaseURL: string,
+    dfnsAuthToken: string,
+    bitcoinNetwork: Network,
+    bitcoinBlockchainAPI: string,
+    bitcoinBlockchainFeeRecommendationAPI: string
+  ) {
+    this.dfnsDelegatedAPIClient = new DfnsDelegatedApiClient({
+      baseUrl: appBaseURL,
+      appId: dfnsAppID,
+      authToken: dfnsAuthToken,
+    });
+    this.bitcoinBlockchainAPI = bitcoinBlockchainAPI;
+    this.bitcoinBlockchainFeeRecommendationAPI = bitcoinBlockchainFeeRecommendationAPI;
+    this.bitcoinNetwork = bitcoinNetwork;
+  }
+
+  private setPayment(fundingPayment: P2TROut, multisigPayment: P2TROut): void {
+    this.payment = {
+      fundingPayment,
+      multisigPayment,
+    };
+  }
+
+  private getPayment(): PaymentInformation {
+    if (!this.payment) {
+      throw new Error('Payment Information not set');
+    }
+    return this.payment;
+  }
+
+  async getWallets(): Promise<ListWalletsResponse> {
+    try {
+      return await this.dfnsDelegatedAPIClient.wallets.listWallets();
+    } catch (error: any) {
+      throw new Error(`Error fetching wallets: ${error}`);
+    }
+  }
+
+  async initializeWalletByID(dfnsWalletID: string): Promise<void> {
+    try {
+      const dfnsWallet = await this.dfnsDelegatedAPIClient.wallets.getWallet({
+        walletId: dfnsWalletID,
+      });
+      this.setDFNSWalletID(dfnsWallet.id);
+      this.setTaprootDerivedPublicKey(dfnsWallet.signingKey.publicKey);
+    } catch (error: any) {
+      throw new Error(`Error fetching wallet: ${error}`);
+    }
+  }
+
+  setDFNSWalletID(dfnsWalletID: string): void {
+    this.dfnsWalletID = dfnsWalletID;
+  }
+
+  getDFNSWalletID(): string {
+    if (!this.dfnsWalletID) {
+      throw new Error('DFNS Wallet ID not set');
+    }
+    return this.dfnsWalletID;
+  }
+
+  setTaprootDerivedPublicKey(taprootDerivedPublicKey: string): void {
+    this.taprootDerivedPublicKey = taprootDerivedPublicKey;
+  }
+
+  getTaprootDerivedPublicKey(): string {
+    if (!this.taprootDerivedPublicKey) {
+      throw new Error('Taproot Derived Public Key not set');
```

Comment:
if these wallets all follow the same format, i wonder if some functions like this could be at the parent class level, and just inherited here?